src/filesystem/impls/filer/FilerFileSystem.js

```diff
@@ -347,35 +347,34 @@ define(function (require, exports, module) {
         });
     }
 
-    function unlink(path, callback) {
-        path = decodePath(path);
-
-        fs.stat(path, function(err, stats) {
+    function _rmfr(path, callback) {
+        // Regardless of whether we're passed a file or dir path, recursively delete it all.
+        fs.rm(path, {recursive: true}, function(err) {
             if (err) {
                 callback(_mapError(err));
                 return;
             }
 
-            // Deal with dir vs. file
-            var fnName = stats.type === "DIRECTORY" ? 'rmdir' : 'unlink';
-            fs[fnName](path, function(err) {
-                // TODO: deal with the symlink case (i.e., only remove cache
-                // item if file is really going away).
-                BlobUtils.remove(path);
+            // TODO: deal with the symlink case (i.e., only remove cache
+            // item if file is really going away).
+            BlobUtils.remove(path);
```

Comment:
Yeah, looks like we never did that for the dir case.  Probably I need to add a way to purge the BlobUtils cache and refresh it by scanning the project root down, since I can't really get a list of all the child entries.